File: app/src/main/java/com/tronography/pixelweather/model/ForecastBuilder.java
package com.tronography.pixelweather.model;

public class ForecastBuilder {
    private double tempMax;
    private double tempMin;
    private String icon;
    private long dateTime;

    public ForecastBuilder setTempMax(double tempMax) {
        this.tempMax = tempMax;
        return this;
    }

    public ForecastBuilder setTempMin(double tempMin) {
        this.tempMin = tempMin;
        return this;
    }

    public ForecastBuilder setIcon(String icon) {
        this.icon = icon;
        return this;
    }

    public ForecastBuilder setDateTime(long dateTime) {
        this.dateTime = dateTime;
        return this;
    }

    public ForecastModel createForecastModel() {
        return new ForecastModel(tempMax, tempMin, icon, dateTime);
    }
}

File: app/src/main/java/com/tronography/pixelweather/model/ForecastBuilder.kt
package com.tronography.pixelweather.model

class ForecastBuilder {
    private var tempMax: Double = 0.toDouble()
    private var tempMin: Double = 0.toDouble()
    private var icon: String? = null
    private var dateTime: Long = 0

    fun setTempMax(tempMax: Double): ForecastBuilder {
        this.tempMax = tempMax
        return this
    }

    fun setTempMin(tempMin: Double): ForecastBuilder {
        this.tempMin = tempMin
        return this
    }

    fun setIcon(icon: String): ForecastBuilder {
        this.icon = icon
        return this
    }

    fun setDateTime(dateTime: Long): ForecastBuilder {
        this.dateTime = dateTime
        return this
    }

    fun build(): ForecastModel {
        return ForecastModel(tempMax, tempMin, icon!!, dateTime)
    }
}